main/src/ca/uwaterloo/flix/language/ast/Symbol.scala

```diff
@@ -178,6 +178,13 @@ object Symbol {
     new TypeAliasSym(ns.parts, ident.name, ident.loc)
   }
 
+  /**
+    * A union type for [[DefnSym]]s and [[SigSym]]s
+    */
+  sealed trait DefnSymOrSigSym {
```

Comment:
Why is this necessary? I don't like this proliferation of multiple inheritance.

Reply:
We have a place that returns either a DefnSym or a SigSym, based on a lookup. I'll see if it can be turned into an enum. Otherwise, more generally would it be useful to have a `Located` trait, so it's more like a type class than a union of types?

Reply:
I replaced the DefOrSig union trait with a disjoint union.